pre-docs/bolt_command_reference.md

```diff
@@ -1,109 +1,498 @@
-# Bolt command reference
+# Bolt Command Reference
 
 Review the subcommands, actions, and options that are available for Bolt.
 
-## Common Bolt commands
+- [apply](#apply)
+- [command run](#command-run)
+- [file upload](#file-upload)
+- [inventory show](#inventory-show)
+- [plan convert](#plan-convert)
+- [plan run](#plan-run)
+- [plan show](#plan-show)
+- [puppetfile install](#puppetfile-install)
+- [puppetfile show-modules](#puppetfile-show-modules)
+- [script run](#script-run)
+- [secret createkeys](#secret-createkeys)
+- [secret decrypt](#secret-decrypt)
+- [secret encrypt](#secret-encrypt)
+- [task run](#task-run)
+- [task show](#task-show)
 
-Bolt commands use the syntax: `bolt <subcommand> <action> [options]`
 
-|Command|Description|Arguments|
-|-------|-----------|---------|
-| `bolt command run` `<COMMAND>` | Runs a command on remote nodes. |- The command, single quoted if it contains spaces or special characters.<br>- The nodes on which to run the command.
-| `bolt script run` | Runs a script in any language that will run on the remote system. |- A path to a local script to run on the remote nodes.<br>- Optionally, arguments to pass to the script.<br>- The nodes on which to run the script.
-| `bolt task run` | Runs a task on a remote system, passing any specified parameters. | - The task name, in the format `modulename::taskname`.<br>- The nodes on which to run the task.
-| `bolt plan run` | Runs a task plan. | - The plan name, in the format `modulename::planname`.<br>- The nodes on which to run the plan.
-| `bolt apply` | Applies a Puppet manifest file. | - The path to the manifest file.<br>- The nodes on which to run the plan.
-| `bolt file upload` | Uploads a local file or directory to a remote node. | - The path to the source file or directory.<br>- The path to the remote location.<br>- The nodes on which to upload the file or directory.
-| `bolt task show` | Lists all the tasks on the modulepath that have not been marked `private`. Will note whether a task supports no-operation mode. | - Adding a specific task name displays details and parameters for the task.<br>- Optionally, the name of a task you want details for: `bolt task show <TASK NAME>`
-| `bolt plan show` | Lists the plans that are installed on the current module path. | - Adding a specific plan name displays details and parameters for the plan.
-| `bolt plan convert` | Converts a YAML plan to a Puppet plan | - The path (relative or absolute) to the YAML plan to be converted.
-| `bolt puppetfile install` | Installs the modules listed in the Puppetfile to the current `Boltdir`. | |
-| `bolt puppetfile show-modules` | Lists the modules, and their versions, that are installed in the current `Boltdir`. | |
-| `bolt inventory show` | Show the list of targets an action would run on. | - The inventory targets to show. |
+## Global Options
 
-## Command options
+The following options are available for all subcommands and actions.
```

Comment:
Nit-picky, but we could slightly shorten this, and it feels like it wants a colon:

`These options are available for all subcommands and actions:`